### PR TITLE
 Several enhancements/fixes

### PR DESCRIPTION
**Pull request by avasquez614**
* Added DeploymentEventsWatcher job that reads the event file of the deployer and does the actual cache clearing/context rebuild.
* Added a global job scheduler and a global thread pool that can be shared by schedulers (therefore decreasing the number of threads created by schedulers).

Ticket craftercms/craftercms#2935

### CODE REVIEW

File: src/main/java/org/craftercms/engine/service/context/SiteContextFactory.java [hunk 1]
```diff
@@ -63,6 +63,7 @@
 import java.io.InputStream;
 import java.net.URLClassLoader;
 import java.util.*;
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/engine/service/context/SiteContextFactory.java]: Please remove and include specifics.

File: src/main/java/org/craftercms/engine/util/deployment/DeploymentEventsWatcher.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
+package org.craftercms.engine.util.deployment;
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/engine/util/deployment/DeploymentEventsWatcher.java]: Missing copyright notice.

File: src/main/resources/crafter/engine/server-config.properties [hunk 3]
```diff
@@ -178,6 +179,15 @@ crafter.security.accessDenied.error.url=/crafter-controller/error/403
 # The XPath queries to extract the roles authorized to access a page
 crafter.security.authorization.page.authorizedRoles.xpathQuery=*/authorizedRoles/item/role
 
+#######################
+# Cron Job Properties #
+#######################
+# The core size of the job thread pool. After this value any new jobs starting up will have a new thread, and
+# any excess threads will be destroyed after the keep alive time
+crafter.engine.jobs.threadPool.coreSize=10
```
Comment (sumerjabri) [inline src/main/resources/crafter/engine/server-config.properties]: Let's discuss.

Comment (avasquez614) [reply src/main/resources/crafter/engine/server-config.properties]: Discussed this and agreed it was ok, just one deployment events watcher job is run for all sites.

**Pull request merged**